helix-rest/src/main/java/org/apache/helix/rest/server/service/InstanceServiceImpl.java

```diff
@@ -230,7 +304,8 @@ protected PartitionHealth generatePartitionHealthMapFromZK() {
     for (int i = 0; i < liveInstances.size(); i++) {
       String instance = liveInstances.get(i);
       // TODO: Check ZNRecord is null or not. Need logic to check whether the healthreports exist
```

Comment:
Will you add this part in separate PR?

Reply:
Yes. There'll be another PRs to address the remaining TODO in the PR. Given the changes already getting so big, I don't want to add more to it. And the changes don't break anything (existing integration test and unit test on single instance stoppable API still works)